src/service/ToolService.ts

```
@@ -256,6 +258,110 @@ export default class ToolService {
 	static schemas: OpenAi.ChatCompletionTool[] = Array.from(ToolService.tools.values()).map((tool) => tool.schema);
+
+	// Helper function to convert JSON schema to Zod schema
+	private static jsonSchemaToZod(schema: any): z.ZodTypeAny {
+		if (!schema) return z.any();
+
+		switch (schema.type) {
+			case 'object': {
+				const shape: Record<string, z.ZodTypeAny> = {};
+				for (const key in schema.properties) {
+					const propSchema = schema.properties[key];
+					let zodProp = this.jsonSchemaToZod(propSchema);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)